### DISCUSSION TITLE
Module not found: Error: Can't resolve 'util' in web-application-quick-start

### DISCUSSION THREAD

**Discussion by neuralmax** [iTwin/itwinjs-core#5428] [Q&A] [answered]
Hello,
I am trying to launch example found in https://developer.bentley.com/tutorials/web-application-quick-start/
and I get an error:
```
Compiled with problems:
ERROR in ./node_modules/semver/classes/semver.js 19:46-69
Module not found: Error: Can't resolve 'util' in 'C:\itwin\web-viewer2\node_modules\semver\classes'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
```
I tried several solutions I managed to google, but they didn't result to anything different.
Please help me to solve this.

**Comment by aruniverse** [comment, marked as answer]
Can you try pinning the dep on `semver` to `7.4.0`?

If you're using npm, you can use [npm-force-resolutions](https://www.npmjs.com/package/npm-force-resolutions)
If you're using yarn, you can use [yarn resolutions](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/)
If you're using pnpm, you can use [pnpm overrides](https://pnpm.io/package_json#pnpmoverrides)

This [semver pr](https://github.com/npm/node-semver/pull/552/files#diff-7917f8ef42ff71df5bde71c259d15feab8b147e3e6c2fde11b56cbf42cadea16R19) introduced a dependency on node polyfills not available on the browser when using webpack5

**Reply by claudiaareneee** [reply]
Also had this issue. This answer helped. For anyone else using npm these are the things I added to fix it: 

In `package.json`:
  ```js title="package.json"
"scripts": {
    "preinstall": "npx npm-force-resolutions",
    /*other scripts*/
  },

  /*other code*/

  "resolutions": {
    "semver": "7.4.0"
  }
```

Then I deleted my `node_modules` and did a fresh `npm install`

**Reply by neuralmax** [reply]
@claudiaareneee Your description is clearer than npm-force-resolutions readme.
Also npm-force-resolutions readme stated that this solution is not secure.
It would be interesting to know how to properly solve this.
